docs/source/advanced/solvePNP.rst

```diff
@@ -5,14 +5,97 @@ SolvePNP
+
+
+Before callibrating:
+
+- Make sure the chessboard is 8 x 8 with the length and width of EACH sqaure is the same and correctly inputted into Chameleon
+- Collect images from different areas, tilts and distances, tilts of +/- 45 degrees is recommended, however, rotating too much may result in inaccurate data 
+- Lighting: make sure no shadows are being cast onto the chessboard, and the chessboard is recieving even lighting
+- Make sure you see rainbow stripes from the corners of the chessboard before you take a snapshot.
+
 .. image:: /images/solvepnp/Thumb.PNG
    :alt: Calibration using the provided chessboard image.
 
+
+After you complete the callibration process, on the command line, you'll see:
+
+- Camera Matrix
+
+   - The camera matrix is a 3 by 3 matrix as show below
+   - .. image:: /images/solvepnp/camera_matrix.PNG
+   - **fx** and **fy** : These are the focal lengths of the camera expressed in pixels
+   - **cx** and **cy** : These are the principal points and refer to the image center 
+
+- Distortion Coefficents
+
+   - distortion coefficents are vectors of up to 8 parameters, formated like : **k1, k2, p1, p2 [, k3 [, k4, k5, k6]]**
+   - **k1, k2..** are radical distortion coefficents
+   - **p1, p2...** are tangential distortion coefficents
+   - The distortion do not change based off the camera resolution, so the distortion coefficents will be the same for 640 x 480 images and 320 x 240 images
+
+Now that we have the camera matrix and distortion coefficents, the final parameter we need is the real world coordinates.
+These coordinates are derived by using the corners of your target in 3D space, following the format:
+
+   - -targetwidth/2, targetheight/2, 0
+   - -targetwidth/2, -targetheight/2, 0
+   - targetwidth/2, -targetheight/2, 0
+   - targetwidth/2, targetheight/2, 0
+
+These world coordinates for the 2020 season are premade for the Loading Bay and Upper port in the dropdown menu or can
+be uploaded as a .csv file.
+
+Now that we have callibrated the camera, and found the 3D world coordinates, we can now use this information to:
+
+- Undisort an image recieved from the camera
+- Using the 3D world coordinates and 2D target coordinates from the image, the SolvePNP() function returns the camera pose using the camera matrix and distortion coefficents
+- Returns a re-projection .. error
+
+OpenCV calculates reprojection error by projecting three-dimensional of chessboard points into the image using the final set of calibration parameters and comparing the position of the corners. An RMS error of 300 means that, on average, each of these projected points is 300 px away from its actual position.
+The closer the error is to 0, the better the callibration. 
+
+Before enabling 3D tracking, calculate the pitch between the horizontal axis and the optical axis of your camera and enter
+it into the **Camera Pitch** field in the chameleon UI.
+
+.. image:: /images/solvepnp/pitch.PNG
+   :alt: Source: Research Gate
```

Comment:
not exactly the point of alt text, and it's very vague as well.

Reply:
Alright took it out then